tests/src/testfebases.cpp

```diff
@@ -0,0 +1,164 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include <config.h>
+
+#include "testhelpers.hh"
+
+#include <dune/common/test/testsuite.hh>
+#include <dune/functions/functionspacebases/basistags.hh>
+#include <dune/functions/functionspacebases/compositebasis.hh>
+#include <dune/functions/functionspacebases/lagrangebasis.hh>
+#include <dune/functions/functionspacebases/lagrangedgbasis.hh>
+#include <dune/functions/functionspacebases/nedelecbasis.hh>
+#include <dune/functions/functionspacebases/powerbasis.hh>
+#include <dune/functions/functionspacebases/raviartthomasbasis.hh>
+#include <dune/grid/yaspgrid.hh>
+
+using Dune::TestSuite;
+
+#include <ikarus/finiteelements/febases.hh>
+#include <ikarus/finiteelements/fehelper.hh>
+#include <ikarus/utils/basis.hh>
+#include <ikarus/utils/init.hh>
+
+/**
+ * \brief A helper function to get a FEBase.
+ *
+ * \tparam GridView Type of the grid view
+ * \tparam PreBasis Type of the pre basis
+ * *
+ * \param gridView The DUNE grid view.
+ * \param preBasis The pre basis to make a basis.
+ *
+ * \return FEBase corresponding to the gridView and preBasis.
+ */
+template <typename GridView, typename PreBasis>
+auto getBasis(const GridView& gridView, const PreBasis& preBasis) {
+  const auto basis      = Ikarus::makeBasis(gridView, preBasis);
+  const auto& localView = basis.flat().localView();
```

Comment:
You can use an imediate evaluated lambda here, because otherwise you can no do this inside a if constexpr branch.

```cpp
const auto& localView = [&]() -> auto& {
if constexpr (useFlat)
   return basis.flat().localView();
else
   return basis.untouched().localView();
}();

Reply:
And all the example bases shoujld run with the non-type template `bool useFlat=false/true`